### ISSUE TITLE
UI improvement request

### ISSUE THREAD

**Issue by CaesarW** [andOTP/andOTP#155]
Hey dear developers!

The app is very useful with its function and being open source. But for user interface there's a little imperfection.

The FAB menu is not very _ Material Design-ish _. The label for menu options and the FAB menu animation still require improving.

Thanks guys for hard work! 👍👍

**Comment by enwokoma** [comment]
Not bad, I've been wanting to make mention of this a long time ago! So, to add to this, 

One major thing I would suggest concerning the FAB is that, the mainlayout of MainActivity should be dimmed/blurred whenever the FAB menu is open, outside this blurred background suggestion, I honestly see no need for any other improvements of the FAB in andOTP because it's not looking bad as it is currently! 

Then another is concerning the animations in andOTP, I think those can be improved too, especially on the Intro Screen, this would help improve the User Experience of andOTP! 

Then, I would suggest that a splash screen should shown on the first run of every new update of andOTP, this Splash Screen should contain a list of What's New in the new update and a text below the screen showing the update version number, this screen should have a nice interactive (not necessarily animated) background and the What's New texts should be written in a nice font! 

Another suggestion is that, we need to reduce the amount of dialog boxes in andOTP (especially for the open source licenses button, I really don't think using a dialog box to display the open source libraries used is necessary as there would probably be more libraries that would be integrated in the near future and a continuous uses of a dialog box to display these licenses would make the dialog box too long and boring to read when more libraries are added to andOTP), so we need to consider using webviews/fragments or anything that feels better (take for instance, I believe that when the open source licenses button is clicked on, a fragment containing the names of all the libraries should be shown, each of those libraries listed should be clickable and when click, it should show another fragment containing the detailed information about that library and how it was used in andOTP) - Well, I think this particular suggestion might not be necessary though! 

Then finally and most importantly, I believe and yet suggest that the QR code Scanner should be updated/changed, we need to use a better UI for the camera and QR code scanning feature! 

In order to implement all these, we might consider the use proguard/dexguard in order to reduce the size of andOTP, this is necessary if we want to maintain the lightweight andOTP (below 10mb)

**Comment by enwokoma** [comment]
I just made a pull request in #156 concerning showing our users an activity that contains a detailed What's New about every new update!

**Comment by flocke** [comment]
Let's just collect all requested changes in a short list:
- [ ] Rework the FAB menu and make it more "material" (I'm probably just gonna replace my own quickly thrown together solution with this: https://github.com/jahirfiquitiva/FABsMenu)
- [ ] (maybe) Replace the currently used dialog to display open source licenses with something nicer
- [ ] (maybe) Make the UI of the QR scanner a bit nicer (since we use [ZXing Android Embedded](https://github.com/journeyapps/zxing-android-embedded), which as far as I know is one of the best QR libraries out there, I will have to see how that could be done)

**Comment by flocke** [comment]
@enwokoma I created a separate branch to experiment a bit with [AboutLibraries](https://github.com/mikepenz/AboutLibraries) instead of the dialog, this is how it looks so far:

![screenshot_andotp dev _20180328-132738](https://user-images.githubusercontent.com/76665/38026429-567f8006-328c-11e8-99b3-87763733c76b.png)
![screenshot_andotp dev _20180328-132746](https://user-images.githubusercontent.com/76665/38026434-580a9398-328c-11e8-817e-1fe39e76a88f.png)

The idea to make it tabbed like this is taken from [Transportr](https://github.com/grote/Transportr).

**Comment by flocke** [comment]
I still need to do some theming, the library cards are always white even on the two dark themes. Apart from that I think I will go with this design in the next release.

**Comment by enwokoma** [comment]
Nice! Not looking bad. Looks like a familiar library I use for one of my
apps (Splanner)

PS: Sorry for the late notice and little silence. It's just that I have
been really occupied this couple of weeks. I just flew in from California
today, I would be free to attend to all my activities on github later this
week or next!

**Comment by flocke** [comment]
Yeah, as far as I can tell AboutLibraries is used in a lot of stuff. That's why I picked it :wink:
